File: Python/make-the-string-great.py
# Time:  O(n)
# Space: O(1)

class Solution(object):
    def makeGood(self, s):
        """
        :type s: str
        :rtype: str
        """
        stk = []
        for ch in s:
            counter_ch = ch.upper() if ch.islower() else ch.lower()
            if stk and stk[-1] == counter_ch:
                stk.pop()
            else:
                stk.append(ch)
        return "".join(stk)

File: C++/make-the-string-great.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
public:
    string makeGood(string s) {
        string result;
        for (const auto& ch : s) {
            char counter_ch = islower(ch) ? toupper(ch) : tolower(ch);
            if (!result.empty() && result.back() == counter_ch) {
                result.pop_back();
            } else {
                result.push_back(ch);
            }
        }
        return result;
    }
};
